plugins/source/azuredevops/codegen/recipes/base.go

```diff
@@ -0,0 +1,114 @@
+package recipes
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"go/format"
+	"os"
+	"path"
+	"reflect"
+	"runtime"
+	"text/template"
+
+	"github.com/cloudquery/plugin-sdk/codegen"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/google/uuid"
+	"github.com/iancoleman/strcase"
+)
+
+//go:embed templates/*.go.tpl
+var templatesFS embed.FS
+
+type Resource struct {
+	Service    string
+	SubService string
+	Struct     interface{}
+	Table      *codegen.TableDefinition
+}
+
+func renderTemplate(name string, filePath string, data interface{}) error {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		return fmt.Errorf("failed to get caller information")
+	}
+	dir := path.Dir(filename)
+
+	tpl, err := template.New(name).Funcs(template.FuncMap{"ToCamel": strcase.ToCamel}).ParseFS(templatesFS, "templates/*.go.tpl")
```

Comment:
posibly extract this tpl parse code into a separate func? will come in handy for mocks.
See https://github.com/cloudquery/cloudquery/blob/main/plugins/source/tailscale/codegen/resources/template.go

Reply:
Would it be easier to do it when we actually need to re-use the function? This PR doesn't add mocks and this code is only used here.